smarts/waymo/waymo_utility.py

```diff
@@ -0,0 +1,2217 @@
+# Copyright (C) 2020. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# pytype: disable=import-error
+
+# Text based Waymo Dataset Browser.
+import argparse
+import copy
+import json
+import os
+import re
+import shutil
+import struct
+import sys
+from pathlib import Path
+from typing import Dict, List, Tuple, Optional, Union, Generator
+from itertools import product
+from multiprocessing import cpu_count
+import matplotlib.pyplot as plt
+import yaml
+from matplotlib.animation import FuncAnimation, FFMpegWriter
+from matplotlib.lines import Line2D
+
+try:
+    import readline
+    from pathos.multiprocessing import ProcessingPool as Pool
+    from tabulate import tabulate
+    from waymo_open_dataset.protos import scenario_pb2
+except (ModuleNotFoundError, ImportError):
+    print(sys.exc_info())
+    print(
+        "You may not have installed the [waymo] dependencies required to run this Waymo Utility. Install them first using the command `pip install -e .[waymo]` at the source directory."
+    )
```

Comment:
Looking at this again I remember that this is supposed to be possible to use without installing smarts proper. I could suggest providing an additional option for how to use this utility.